knockoff/utilities/io.py

```diff
@@ -0,0 +1,41 @@
+import logging
+
+from sqlalchemy import create_engine
+from joblib import Parallel, delayed
+
+logger = logging.getLogger(__name__)
+
+
+def _to_sql(df, table, url, **kwargs):
+    to_sql_kwargs = {
+        'index': False,
+        'method': 'multi',
+        'if_exists': 'append'
+    }
+    to_sql_kwargs.update(kwargs)
+    engine = create_engine(url)
+    with engine.connect() as conn:
+        df.to_sql(table, conn, **to_sql_kwargs)
+
+
+def to_sql(df,
+           table,
+           url,
+           parallelize=True,
+           chunksize=1000,
+           n_jobs=-1,
+           **kwargs):
+    logger.info("Populating table: {}".format(table))
+    # TODO: better default for more effect parallelization?
+    nrows = df.shape[0]
+    if parallelize and nrows > chunksize:
+        Parallel(n_jobs=n_jobs)(
```

Comment:
Does the table already exists when the inserts are happening?
I did run into an edge case where the table did not exist and multiple threads/processes tried to create it. I had to use lock if creating a new table and writing to it.

Reply:
No assumptions are made here as to whether the tables exists or not, but I imagine most use-cases will be populating pre-existing tables (potentially through the TempDatabaseService)